telink: Factory Data sample

interfer opened the pull request:
Read, parse and print Matter Factory Data Partition from simple Zephyr sample.
You should have factory_partition present/flashed and run the sample after.

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/src/factory_data_parser.c:
```diff
@@ -0,0 +1,349 @@
+
+const struct device *mFlashDevice = DEVICE_DT_GET(DT_CHOSEN(zephyr_flash_controller));
+
+__attribute__((section(
```
What is the reason to keep RAM image?
As I understand it's enough to declare structure. And make const pointer to flash area.

interfer replied:
@andriy-bilynskyy Refactored now

interfer committed:
telink: samples: FactoryData verifier for Telink

Read, parse and print Matter Factory Data Partition
from simple Zephyr sample.
You should have factory_partition present/flashed
and run the sample after.

telink: samples: refactored FactoryData parser

Got rid of flash reading operation. Now direct
copying from flash memory address via pointer.

telink: samples: CI build for B92 FactoryData

Added Factory Data verifier CI job to B92 build workflow.
Removed obsolete headers, used zcbor_string instead of custom created.

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/src/factory_data_parser.h:
```diff
@@ -0,0 +1,78 @@
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/storage/flash_map.h>
```
Why this headers are here?

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/src/factory_data_parser.h:
```diff
@@ -0,0 +1,78 @@
+#include <stdint.h>
+
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/kernel.h>
```
Same...

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/src/factory_data_parser.c:
```diff
@@ -0,0 +1,302 @@
+			break;
+		}
+
+		if (strncmp("version", (const char *)currentString.value, currentString.len) == 0) {
```
memcmp

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/prj.conf:
```diff
@@ -0,0 +1,2 @@
+CONFIG_FLASH=y
+CONFIG_ZCBOR=y
```
seems CONFIG_LOG?

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/src/main.c:
```diff
@@ -0,0 +1,34 @@
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
```
or why log is here?

andriy-bilynskyy commented:
Sqush your commits in PR

andriy-bilynskyy commented on samples/boards/tlsr9x/factorydata/src/factory_data_parser.h:
```diff
@@ -0,0 +1,78 @@
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/kernel.h>
+
+struct FactoryDataString {
```
struct zcbor_string???

andriy-bilynskyy commented:
Please add ci for target you tested
